src/platformio.ts

```diff
@@ -100,6 +120,37 @@ export class PlatformIO {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private configureAutoUpdateIncludes(): void {
+        const autoUpdateIncludes = vscode.workspace.getConfiguration("platformio").get<boolean>("autoUpdateIncludes");
+        if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+            const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**");
+            this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+            fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            const iniFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/platformio.ini");
+            this._autoUpdateIncludesDisposables.push(iniFileWatcher);
+            iniFileWatcher.onDidChange((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+            iniFileWatcher.onDidCreate((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+        }
+
+        if (!autoUpdateIncludes && this._autoUpdateIncludesEnabled) {
+            this.disposeAutoUpdateIncludes();
+        }
+
+        this._autoUpdateIncludesEnabled = autoUpdateIncludes;
+    }
+
+    private fileSystemDidChange(e: vscode.Uri) {
+        clearTimeout(this._debounceTimer);
+        this._debounceTimer = setTimeout(() => this.addIncludePath(true), 5000);
+    }

```

Comment:
I didn't even realize that you could use .ino files, so I didn't think about testing that, but it makes me realize that there is also a problem with the .vscode/BROWSE.VC.DB* files and the .pioenvs folder, so I have also added them to the _excludePaths.